admin-dev/themes/new-theme/js/pages/currency/form/currency-form.js

```diff
@@ -96,36 +141,60 @@ export default class CurrencyForm {
     this._resetCurrencyData(this.$isoCodeInput.val());
   }
 
-  _resetCurrencyData(selectedISOCode) {
+  async _resetCurrencyData(selectedISOCode) {
     this.$loadingDataModal.modal('show');
     this.$resetDefaultSettingsButton.addClass('spinner');
-    const getCurrencyData = this.getCLDRDataUrl.replace('CURRENCY_ISO_CODE', selectedISOCode);
-    $.get(getCurrencyData)
-      .then((currencyData) => {
-        for (let langId in currencyData.names) {
-          let langNameSelector = this.map.namesInput(langId);
-          $(langNameSelector).val(currencyData.names[langId]);
-        }
-        for (let langId in currencyData.symbols) {
-          let langSymbolSelector = this.map.symbolsInput(langId);
-          $(langSymbolSelector).val(currencyData.symbols[langId]);
-        }
-        this.$isoCodeInput.val(currencyData.isoCode);
-        if (currencyData.exchangeRate) {
-          this.$exchangeRateInput.val(currencyData.exchangeRate);
-        }
-        this.$precisionInput.val(currencyData.precision);
-      })
-      .fail((currencyData) => {
-        let errorMessage = 'Can not find CLDR data for currency ' + selectedISOCode;
-        if (currencyData && currencyData.responseJSON && currencyData.responseJSON.error) {
-          errorMessage = currencyData.responseJSON.error;
-        }
-        showGrowl('error', errorMessage, 3000);
-      })
-      .always(() => {
-        this.$loadingDataModal.modal('hide');
-        this.$resetDefaultSettingsButton.removeClass('spinner');
-      })
+
+    await this.store.dispatch('fetchReferenceData', selectedISOCode);
```

Comment:
We shouldn't mix vuejs and vanilla

Reply:
`data` can be passed to the master vue component, and in this one, you can use vuejs lifecycle events to load everything you want from the store.
The logic & events must stay as must as possible in the VueJs context.
Will be really easier to refactor, move, and delete this bad way they implements vue :(